.github/workflows/pull-e2e-upgrade-test.yaml

```diff
@@ -0,0 +1,93 @@
+name: pull-e2e-upgrade-test
+
+env:
+  DOCKER_IMAGE: europe-docker.pkg.dev/kyma-project/dev/nats-manager:PR-${{ github.event.number }}
+  E2E_LOG_LEVEL: debug
+  KYMA_STABILITY: "unstable"
+  KYMA: "./hack/kyma"
+
+on:
+  pull_request:
+    branches: [ "main" ]
+    paths-ignore:
+      - 'docs/**'
+      - '**.md'
+      - 'sec-scanners-config.yaml'
+
+jobs:
+  e2e-upgrade: # This job tests the upgrade of NATS module from latest image of the main branch to the current commit.
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Install k3d tools
+        run: |
+          make -C hack/ci/ install-k3d-tools
+
+      - name: Install Kyma CLI & setup k3d cluster using kyma CLI
+        run: |
+          make kyma
+          make -C hack/ci/ create-k3d
+          kubectl version
+          kubectl cluster-info
+
+      - name: Install latest NATS manager from main branch
+        run: |
+          make -C hack/ci/ create-kyma-system-ns
+          make deploy IMG=europe-docker.pkg.dev/kyma-project/prod/nats-manager:latest
```

Comment:
Please make sure that the latest tag refers to main head. Otherwise we can use git to get the last commit SHA from the main branch.

Reply:
yes, I am sure. I've checked it and it is also set [here](https://github.com/kyma-project/test-infra/blob/cc8b22887d8f54659814a746790a85fb82267791/prow/jobs/kyma-project/nats-manager/nats-manager-generic.yaml#L231)